includes/data-stores/class-wc-admin-reports-orders-stats-data-store.php

```diff
@@ -527,20 +528,24 @@ protected static function is_returning_customer( $order ) {
 			return false;
 		}
 
-		$first_order = $oldest_orders[0];
+		$first_order       = $oldest_orders[0];
+		$second_order      = isset( $oldest_orders[1] ) ? $oldest_orders[1] : false;
+		$excluded_statuses = self::get_excluded_report_order_statuses();
 
 		// Order is older than previous first order.
-		if ( $order->get_date_created() < new WC_DateTime( $first_order->date_created ) ) {
+		if ( $order->get_date_created() < new WC_DateTime( $first_order->date_created ) &&
+			! in_array( $order->get_status(), $excluded_statuses, true )
+		) {
 			self::set_customer_first_order( $customer_id, $order->get_id() );
 			return false;
 		}
-		// First order date has changed and next oldest is now the first order.
-		$second_order = isset( $oldest_orders[1] ) ? $oldest_orders[1] : false;
+		// First order date or status has changed and next oldest is now the first order.
+		$date_change   = $second_order && $order->get_date_created() > new WC_DateTime( $first_order->date_created ) &&
+			new WC_DateTime( $second_order->date_created ) < $order->get_date_created();
```

Comment:
Nitpick: I think this would be easier to read if each condition was on its own line with a comment.

Reply:
Agreed.  Refactored in https://github.com/woocommerce/wc-admin/pull/1557/commits/ba3f70edc42ad4e4957b401a2931d99b9f5a2261